CONTRIBUTING.md

```diff
@@ -2,33 +2,79 @@
 
-# Code Guidelines
+## Code Guidelines
 
 We make a best-effort attempt to adhere to [PEP 8 - Style Guide for Python Code](https://www.python.org/dev/peps/pep-0008/).  Keep the following context in mind:
 * Our default stance is to run linter checks during the build process.
 * Notable exception:  [naming conventions](https://peps.python.org/pep-0008/#naming-conventions).
 
-## Naming Convention Guidelines
+### Naming Convention Guidelines
 Since PyTeal aims for backwards compatibility, it's _not_ straightforward to change naming conventions in public APIs.  Consequently, the repo contains some deviations from PEP 8 naming conventions.
 
 In order to retain a consistent style, we prefer to continue deviating from PEP 8 naming conventions in the following cases.  We try to balance minimizing exceptions against providing a consistent style for existing software.
 * Enums - Define with lowercase camelcase.  Example:  https://github.com/algorand/pyteal/blob/7c953f600113abcb9a31df68165b61a2c897f591/pyteal/ast/txn.py#L37
 * Factory methods - Define following [class name](https://peps.python.org/pep-0008/#class-names) conventions.  Example:  https://github.com/algorand/pyteal/blob/7c953f600113abcb9a31df68165b61a2c897f591/pyteal/ast/ternaryexpr.py#L63
 
 Since it's challenging to enforce these exceptions with a linter, we rely on PR creators and reviewers to make a best-effort attempt to enforce agreed upon naming conventions.
+
+### Module Guidelines
+
+Every directory containing source code should be a Python module, meaning it should have an `__init__.py` file. This `__init__.py` file is responsible for exporting all public objects (i.e. classes, functions, and constants) for use outside of that module.
+
+Modules may be created inside of other modules, in which case the deeper module is called a submodule or child module, and the module that contains it is called the parent module. For example, `pyteal` is the parent module to `pyteal.ast`.
+
+A sibling module is defined as a different child module of the parent module. For example, `pyteal.ast` and `pyteal.ir` are sibling modules.
+
+### Import Guidelines
+
+#### In Runtime Code
+
+When a runtime file in this codebase needs to import another module/file in this codebase, you should import the absolute path of the module/file, not the relative one, using the `from X import Y` method.
```

Comment:
Do you want to also allow for the possibility of importing the module without `from`, e.g. `mypkg.sibling`  and `mypkg.sibling as sister`? (I got the first example from [pep-8](https://peps.python.org/pep-0008/#imports))

Reply:
For simplicity, I'd say _no_, let's not allow that, but I'm open to revisiting if there's a motivating use case.

Reply:
One motivation is for the tests use case where we use `import pyteal as pt` or in general, or in general, importing many objects like `from X import Y1, Y2, Y3, .... , Y30` and it gets pretty gnarly. 
There is one obligatory use case: avoiding namespace collisions as was mentioned in the PEP itself.